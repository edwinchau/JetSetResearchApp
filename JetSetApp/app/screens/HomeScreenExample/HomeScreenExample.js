import React, {Component, Fragment} from 'react';
import { StyleSheet, Text, View, Button} from 'react-native';
import moment from 'moment';

import styles from './Styles'
import SaveData from '../../resources/SaveData/SaveData';
import SendData from '../../resources/SendData/SendData';

export default class HomeScreenExample extends Component {
    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: '#fff',
            headerTitle: 'JetSet',
            headerTitleStyle: {
                flex: 1,
            }
        };
    };

    constructor(props) {
        super(props);
        this.state = { time: Date.now() };
    }

    componentDidMount() {
        this.interval = setInterval(() => this.setState({ time: Date.now() }), 1000);
    }
    componentWillUnmount() {
        clearInterval(this.interval);
    }

    render() {
        const { navigate } = this.props.navigation;

        let survey;

        if (parseInt(moment(this.state.time).format('ss')) % 2 === 0) {
            survey = (<Fragment>
                <Button
                    onPress={() => navigate('Survey', { survey: 'BreakfastQuestions' })}
                    title="Breakfast"
                />

                <Button
                    onPress={() => navigate('Survey', { survey: 'LunchQuestions' })}
                    title="Lunch"
                />

                <Button
                    onPress={() => navigate('Survey', { survey: 'DinnerQuestions' })}
                    title="Dinner"
                />
            </Fragment>)
        } else {
            survey = (<Fragment>
                <Button
                    onPress={() => navigate('Survey', { survey: 'BreakfastQuestions' })}
                    title="Breakfast 1"
                />

                <Button
                    onPress={() => navigate('Survey', { survey: 'LunchQuestions' })}
                    title="Lunch 1"
                />

                <Button
                    onPress={() => navigate('Survey', { survey: 'DinnerQuestions' })}
                    title="Dinner 1"
                />
            </Fragment>)
        }

        return (
            <View style={styles.background}>
                <View style={styles.container}>
                    <Text>Time: {moment(this.state.time).format()}</Text>
                </View>
                <View style={styles.container}>

                    {survey}

                </View>
                <View style={styles.container}>
                    <Button
                        onPress={() => SendData.sendDataEmail()}
                        title="Send Data"
                    />
                </View>
                <View style={styles.container}>
                    <Text>Testing Features</Text>

                    <Button
                        onPress={() => SaveData.displayAllFiles()}
                        title="View Files"
                    />

                    <Button
                        onPress={() => SaveData.deleteFile("NewUserQuestions.json")}
                        title="Delete User Save"
                    />
                </View>
            </View>
        );
    }
}
