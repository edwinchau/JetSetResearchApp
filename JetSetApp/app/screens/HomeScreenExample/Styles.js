import { StyleSheet } from 'react-native';

let styles = StyleSheet.create({
    background: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'white',
    },
    container: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'white',
        elevation: 20,
        borderRadius: 15,
        paddingTop: 10,
        paddingBottom: 10,
    },
    breakfast: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'rgba(244,204,112,1)',
        elevation: 20,
        borderRadius: 15,
        paddingTop: 13,
        paddingBottom: 13,
        marginTop: 10,
        marginBottom: 10,
    },
    lunch: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'rgba(222,122,34,1)',
        elevation: 20,
        borderRadius: 15,
        paddingTop: 13,
        paddingBottom: 13,
        marginTop: 10,
        marginBottom: 10,
    },
    dinner: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'rgba(32,148,139,1)',
        elevation: 20,
        borderRadius: 15,
        paddingTop: 13,
        paddingBottom: 13,
        marginTop: 10,
        marginBottom: 10,
    },
    admin: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'rgba(106,177,135,1)',
        elevation: 20,
        borderRadius: 15,
        paddingTop: 13,
        paddingBottom: 13,
        marginTop: 10,
        marginBottom: 10,
    },
    other: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'rgba(106,177,135,1)',
        elevation: 20,
        borderRadius: 15,
        marginBottom: 10,
    },
    questionText: {
        marginBottom: 20,
        fontSize: 30,
    },
    headerStyle: {
        backgroundColor: 'rgba(106,177,135,1)',
        height: 40,
        elevation: 5,
    },
});

module.exports = styles;
