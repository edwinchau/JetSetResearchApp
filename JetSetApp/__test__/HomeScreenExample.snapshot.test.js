import React from "react";
import HomeScreenExample from "../app/screens/HomeScreenExample/HomeScreenExample"
import renderer from "react-test-renderer"

describe("<HomeScreenExample/>", ()=>{
  it("snapshot", ()=>{
    expect(renderer.create(<HomeScreenExample/>).toJSON()).toMatchSnapshot();
    
  });
});
