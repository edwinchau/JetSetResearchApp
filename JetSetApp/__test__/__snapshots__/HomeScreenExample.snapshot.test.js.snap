// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreenExample/> snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(108,48,237,1)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <Text>
      Time: 
      2019-10-07T22:52:20+11:00
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Breakfast
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Lunch
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Dinner
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Send Data
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Admin Page
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <Text>
      Testing Features
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          View Files
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Delete User Save
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Notifications permission
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Start scheduled notifications
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Start during flight notifications
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 20,
        "justifyContent": "center",
        "maxWidth": "90%",
        "minWidth": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Cancel all notifications
        </Text>
      </View>
    </View>
  </View>
</View>
`;
